lib/std/os.zig

```diff
@@ -3169,3 +3169,10 @@ pub fn dn_expand(
     }
     return error.InvalidDnsPacket;
 }
+
+pub fn sched_yield() void {
+    switch (builtin.os) {
+        .windows => _ = windows.kernel32.SwitchToThread(),
+        else => assert(system.sched_yield() == 0),
```

Comment:
Maybe instead of asserting just ignore the result?

Reply:
If `sched_yield` is defined to always succeed, which it is on Linux, then `assert` is correct here. If it can possibly fail, then we need to have that reflected in the error set.

Reply:
Is this because sched_yield error'ing out means it will always error out [like in FreeBSD's case](https://www.freebsd.org/cgi/man.cgi?query=sched_yield&sektion=2#end)?

Reply:
Nice, OK so we have to treat this as a fallible API now. That's fine. I propose to make the error set:

```zig
pub const SchedYieldError = error{
    /// The system is not configured to allow yielding
    SystemCannotYield,
};

pub fn sched_yield() SchedYieldError!void {
    if (builtin.os == .windows) {
        // The return value has to do with how many other threads there are; it is not
        // an error condition on Windows.
        _ = windows.kernel32.SwitchToThread();
        return;
    }
    switch (errno(system.sched_yield()) {
        0 => return,
        ENOSYS => return error.SystemCannotYield,
        else => return error.SystemCannotYield,
    }
}
```

Normally we have `Unexpected` as another possibility, but in this case it's unclear why this distinguishing would be beneficial.

Having the `ENOSYS` and `else` on different lines here is useful when looking at an [error return trace](https://ziglang.org/documentation/master/#Error-Return-Traces).

The callsite of `os.sched_yield` can now choose how to deal with this possible error. Probably by ignoring it. But that's why we have these 2 layers of abstraction; the os.zig file is pretty low level, and exposes all the functionality of the underlying syscall, including errors.